File: eShop-BackEnd/src/main/java/co/com/icesi/eShopBackEnd/model/Order.java
package co.com.icesi.eShopBackEnd.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;
import java.util.UUID;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Order {

    @Id
    private UUID orderId;
    private String status;
    private Long total;

    @ManyToOne
    @JoinColumn(name = "user_user_id")
    private User user;

    @ManyToMany(cascade = { CascadeType.ALL })
    @JoinTable(
            name = "order_item",
            joinColumns = { @JoinColumn(name = "order_id") },
            inverseJoinColumns = { @JoinColumn(name = "item_id") }
    )
    private List<Item> items;

}

File: eShop_Front/src/interfaces/Order.ts
export interface Order {
    orderId: string;
    user: string;
    status: string;
    total: string;
    items: Array<string>;
}

export interface ResponseOrderDTO {
    orderId: string;
    user: string;
    status: string;
    total: number;
    items: ResponseItemDTO[];
  }
  
  export interface ResponseItemDTO {
    itemId: string;
    name: string;
    description: string;
    price: number;
    imageUrl: string;
    category: string | null;
  }